spec/code_block_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'spec_helper'
+require 'kramdown'
+require 'ap'
+require 'pp'
+site = File.join(File.dirname(__FILE__), '..','content', '**', '*.md')
+PAGES = Dir.glob(site)
+
+def extract_code(element,prev_element)
+  codes = [] 
+  if element.type == :codespan \
+    && !element.value.nil? \
+    && !codes.include?(element.value) \
```

Comment:
Actually, I'd prefer to include even the already existing codes and only avoid executing / checking the same one multiple times.
That's because it preserves the `context` information we may need in other tests(related with the comment above).